### DISCUSSION TITLE
Need help dynamically changing the Open-AI model.

### DISCUSSION THREAD

**Discussion by kernelism** [vanna-ai/vanna#311] [Q&A] [answered]
I was wondering whats the best way to do this. I want the user to be able to modify the model through the UI. I notice that the vanna object binded to the open-ai config. I want to change it without having to initialise and connect to db again.

**Comment by zainhoda** [comment, marked as answer]
Depending on your use case if you're using OpenAI, you can do this:

```python
vn.config["model"] = 'gpt-4-turbo-preview'
```

This can be done anytime

**Reply by aitoehigie** [reply]
Can I use gpt-4 without a separate openai subscription?
